core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -445,8 +445,9 @@ class ProducerStateManager(val topicPartition: TopicPartition,
             lastMapOffset = lastSnapOffset
             return
           } catch {
-            case e: CorruptSnapshotException =>
-              error(s"Snapshot file at ${file.getPath} is corrupt: ${e.getMessage}")
+            case e: Exception =>
```

Comment:
Thinking about this a little bit more, I'm not sure it's a good idea to just catch all exceptions blindly and try to continue. For example, if it is an IOException, there could be a more severe problem. Maybe we should content ourselves with catching only invalid schemas and checksum failures?